compiler/src/main/kotlin/se/ansman/kotshi/MethodSpecExt.kt

```diff
@@ -32,7 +32,10 @@ inline fun MethodSpec.Builder.addSwitchBranch(branch: String,
                                               block: MethodSpec.Builder.() -> Unit): MethodSpec.Builder {
     beginControlFlow("$branch:", *args)
     block()
-    addStatement("break")
+    // We explicitly continue instead of break because the switch is the sole statement inside of
+    // a loop. Doing a break forces a jump to the end of the switch and then a jump back up to the
+    // loop conditional.
+    addStatement("continue")
```

Comment:
Should this really be done here unconditionally? What if you use a switch without a loop? Could you add an argument that enables this behaviour?

Reply:
Updated